### DISCUSSION TITLE
Image preview on mobile device

### DISCUSSION THREAD

**Discussion by derritter88** [searxng/searxng#3175] [Ideas]
If searching for images on a mobile browser (in my case iPhone + Firefox) the images are kind of useless previews compared to Duckduckgo which I previously used as the images are not being displayed properly (see screenshots).

![2024-01-31 20-37-37-0](https://github.com/searxng/searxng/assets/8606313/f4a40e1f-e53b-4fbb-8191-e3097225bb62)
![2024-01-31 20-38-08-0](https://github.com/searxng/searxng/assets/8606313/9af9a281-3841-4ac2-9afc-ef2a1ef21b16)

I think at would be a better approach to display thumbnails Duckduckgo style instead of current one.

**Comment by return42** [comment]
What is the resolution of your device?

**Comment by derritter88** [comment]
2532 x 1170 Pixel (iPhone 14 standard)

**Reply by return42** [reply]
Can't believe / the screenshots above are from a smaller device (not more than 300px width).

**Reply by derritter88** [reply]
They were made via my iPhone

**Reply by vojkovic** [reply]
> Can't believe / the screenshots above are from a smaller device (not more than 300px width).

The search logo disappears if the horizontal resolution is less than 320px, so this can't be the case. Looks like a strange error, although I'm unable to reproduce it.

**Comment by Squidlabs2** [comment]
I'm running into the same issue with ios devices as well.   Was there anything resolved from this?   Android and desktop PC's work fine.

**Comment by rune1979** [comment]
Same here, iPhone 11 (SE).

**Comment by kebeabap** [comment]
Tested this on latest firefox / safari on iPhone SE 2022 iOS 18.0.1 and on appetize.io on different devices. Everywhere cropping instead of scaling is applied. Whereas testing on a mobile android device, the images are scaled correctly
Had the same result on every instance I've tested.
Horizontal view sometimes puts multiple images in a row, however crops them as well, same goes for those images which are filling the whole width. 
![searxng-images-ios-horizontal](https://github.com/user-attachments/assets/4246d314-1733-40d6-8e9f-a7c7d62a2cbb)
![searxng-images-ios-portrait](https://github.com/user-attachments/assets/ebfad1f9-13a0-4907-81cf-afbf2e5a0cdd)

**Reply by return42** [reply]
@kebeabap the issue you reported should be fixed in https://github.com/searxng/searxng/pull/3921 .. if you have the ability, please give this PR a try and report your experience there / thanks :+1:

**Comment by ha1dfo** [comment]
With #3921 merged into main branch, the pics show up even narrower. Using version `2024.10.15+058a07240`

The results are like this on a small window (ie: phone):

![image](https://github.com/user-attachments/assets/56091033-0fab-46cf-927d-0b30a1ad34f5)

Same query, same window size on a different search engine:
![image](https://github.com/user-attachments/assets/c098e3eb-fc09-444e-ad76-88f6a3402897)

expected behavior: pics to have their full height proportional to their width matching the browser window size.

**Reply by return42** [reply]
@ha1dfo thanks for your report .. there is already a new issue for :

- https://github.com/searxng/searxng/issues/3928
